### PR TITLE
Home Page

### PR DESCRIPTION
**Pull request by MauroGuerrero20**
# Features
- Creates a Home Page for the website.
  - Includes all of the User's clubs (owner/member)
  - Includes all of the User's booklists (owner/collaborator)

## Minor Features
- Adds Spinners to @steven-solar's ```ClubPage``` and ```MyClubs```.
- Refactor Styling for ClubPage. 
- Viwer mode for SearchUserModal.
  - If the User is a viewer (member/collaborator), then the User can view other Collaborator/Member in the Modal but will not be able to add more. 

# Demo
HomePage Demo

# Demo 2
Landing Page Demo 2

### CODE REVIEW

File: capstone-backend/src/test/java/com/google/sps/servlets/SearchServletTest.java [hunk 1]
```diff
@@ -109,18 +109,6 @@ public boolean checkPopulatedVolumeDataObject(VolumeData book) {
         && book.getWebReaderLink() != null;
   }
 
-  @Test
-  public void ensureJsonResponseIsInCorrectFormat() {
```
Comment (steven-solar) [inline capstone-backend/src/test/java/com/google/sps/servlets/SearchServletTest.java]: Why was this removed?

File: capstone-frontend/src/components/ClubGridItem.jsx [hunk 2]
```diff
@@ -17,22 +17,22 @@ class ClubGridItem extends Component {
...
-      .then(this.setState({ requested: true }))
-      .catch(function (err) {
-        //TODO #61: Centralize error output
-        alert(err);
-      });
+    fetch(`/api/clubs`, {method: "put", body: JSON.stringify(putBody)})
+        .then(this.setState({requested: true}))
+        .catch(function(err) {
+          //TODO #61: Centralize error output
+          alert(err);
```
Comment (steven-solar) [inline capstone-frontend/src/components/ClubGridItem.jsx]: We should remove alerts. Could be in a later PR.

File: capstone-frontend/src/components/ClubGridItem.jsx [hunk 3]
```diff
@@ -17,22 +17,22 @@ class ClubGridItem extends Component {
...
     const putBody = {
       "id": this.props.club.id,
       "remove_requestIDs": window.localStorage.getItem("userID")
     };
-    fetch(`/api/clubs`, { method: "put", body: JSON.stringify(putBody)})
-        .then(this.setState({ requested: false }))
-        .catch(function (err) {
+    fetch(`/api/clubs`, {method: "put", body: JSON.stringify(putBody)})
+        .then(this.setState({requested: false}))
+        .catch(function(err) {
```
Comment (steven-solar) [inline capstone-frontend/src/components/ClubGridItem.jsx:35]: Same here. Up to you if you want to do now or later

File: capstone-frontend/src/components/ClubPage.jsx [hunk 4]
```diff
@@ -19,6 +20,7 @@ class ClubPage extends Component {
       assignments: [],
       owner: {},
       members: [],
+      fetchingData: false, // Spinner
```
Comment (steven-solar) [inline capstone-frontend/src/components/ClubPage.jsx:23]: I like the spinner on this page! Makes for much better UX

File: capstone-frontend/src/components/ClubPage.jsx [hunk 5]
```diff
@@ -122,46 +134,85 @@ class ClubPage extends Component {
...
+      <div >
+
+        {this.state.fetchData ?
+          (<div className="text-center mt-4">
+            <Spinner variant="primary" animation="border" role="status" />
+          </div>)
+          :
+          (
+            <>
+              <Row>
```
Comment (steven-solar) [inline capstone-frontend/src/components/ClubPage.jsx:145]: Why wrap the `Row` in a `React.Fragment`? Isn't the fragment only used when you don't have one parent to hold everything in the return? In this case the Row serves that purpose

File: capstone-frontend/src/components/ClubPage.jsx [hunk 6]
```diff
@@ -122,46 +134,85 @@ class ClubPage extends Component {
...
         }
-        <Row className='justify-content-center'>
-          {members}
-        </Row>
       </div>
     );
   }
 }
 
+
```
Comment (steven-solar) [inline capstone-frontend/src/components/ClubPage.jsx]: `nit`: extra newline

File: capstone-frontend/src/components/CreateList.jsx [hunk 7]
```diff
@@ -223,7 +224,7 @@ class CreateList extends Component {
 
                       <Col md={3} className="px-2 my-0 border" key={addedBook.id}>
                         <BookDescriptionOverlay book={addedBook}>
-                          <img className="img-responsive mt-3 p-0 rounded book-img-sm" src={addedBook.thumbnailLink} alt={addedBook.title} />
+                          <img className="img-fluid mt-3 p-0 rounded book-img-sm" src={addedBook.thumbnailLink} alt={addedBook.title} />
```
Comment (steven-solar) [inline capstone-frontend/src/components/CreateList.jsx:227]: What does this change do?

File: capstone-frontend/src/components/Home.jsx [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+  fetchClubs = async (type) => {
+
+    const userID = window.localStorage.getItem('userID');
+
+    if (type === 'own') {
+      this.setState({ fetchingClubsOwned: true });
+      const clubsOwned = await fetch(`/api/clubs?ownerID=${userID}`).then(resp => resp.json());
+      this.setState({ clubsOwned, fetchingClubsOwned: false });
+    }
+    else if (type = 'shared') {
```
Comment (steven-solar) [inline capstone-frontend/src/components/Home.jsx]: Should this be `===`?

File: capstone-frontend/src/components/Home.jsx [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+      }
+
+      let gBooks = [];
+      let bookIDs;
+
+      // Reduce to one element if on mobile
+      if (this.state.screenWidth <= 425) {
+        bookIDs = bookList.gbookIDs.slice(0, 1);
+      }
+      // Reduce to three elements if on table
```
Comment (steven-solar) [inline capstone-frontend/src/components/Home.jsx]: `nit`: tablet

File: capstone-frontend/src/components/Home.jsx [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+      else if (this.state.screenWidth >= 426 && this.state.screenWidth <= 768) {
+        bookIDs = bookList.gbookIDs.slice(0, 3);
+      }
+      // Reduce to five elements
+      else {
+        bookIDs = bookList.gbookIDs.slice(0, 5);
+      }
```
Comment (steven-solar) [inline capstone-frontend/src/components/Home.jsx:146]: nice job thinking about different screen sizes!

File: capstone-frontend/src/components/Home.jsx [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+      this.setState({ sharedBookListsMap: bookListsMap, fetchingBookListsShared: false });
+    }
+  }
+
+
+  render() {
+    return (
+      <div>
+
+        <Tabs
```
Comment (steven-solar) [inline capstone-frontend/src/components/Home.jsx:171]: I like the Tabs!

File: capstone-frontend/src/components/Home.jsx [hunk 12]
```diff
@@ -0,0 +1,304 @@
...
+                {this.state.clubsOwned.length > 0 ?
+                  <CardDeck className='groups-list-container'>
+                    {this.state.clubsOwned.map(club =>
+                      <ClubGridItem key={club.id} club={club} />)}
+                  </CardDeck>
+                  :
+                  <div className='text-center'>
+                    <h3 className=' mt-5'>No Clubs Found</h3>
+                    <Link to='/createclub'>
+                      <Button className='mt-4' variant='primary'>
```
Comment (steven-solar) [inline capstone-frontend/src/components/Home.jsx:192]: Awesome including this button. Should it be in both clubs tabs? 

Also, we can remove the `MyClubs` page in this PR as well, since it is now covered in our `HomePage`

File: capstone-frontend/src/components/ListPage.jsx [hunk 13]
```diff
@@ -41,14 +41,14 @@ class ListPage extends Component {
...
-    await this.fetchBooks();
+  componentDidMount() {
+    this.fetchBooks();
   }
 
-  async componentDidUpdate(prevProps) {
+  componentDidUpdate(prevProps) {
     if (this.props.match.params.id !== prevProps.match.params.id) {
       this.setState({ loading: true, gBooks: [], bookList: {} })
-      await this.fetchBooks();
```
Comment (steven-solar) [inline capstone-frontend/src/components/ListPage.jsx:51]: nice fixing this since it didn't change any functionality

File: capstone-frontend/src/components/SearchUserModal.jsx [hunk 14]
```diff
@@ -238,75 +238,80 @@ export class SearchUserModal extends Component {
...
+                              <Card.Body>
+                                <Card.Title>
+                                  {user.fullName}
+                                </Card.Title>
+                                <Card.Text id='email-text'>
+                                  {user.email}
+                                </Card.Text>
+
+
+                                {(this.arrayContainsJSONId(this.state.addedUsers, user) &&
```
Comment (steven-solar) [inline capstone-frontend/src/components/SearchUserModal.jsx:287]: Nice way to store the whole user object in state instead of just the id.

**Review by steven-solar — APPROVED** [review]
Looks good! This ties up a lot of loose ends. I would remove the MyClubs page in this PR as well, since it no longer serves any real purpose. Maybe change the `LeftSideBar` link for Clubs to drop down the way it currently does for lists.

File: capstone-frontend/src/components/ClubGridItem.jsx [hunk 15]
```diff
@@ -17,22 +17,22 @@ class ClubGridItem extends Component {
       id: this.props.club.id,
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/ClubGridItem.jsx:17]: I feel like these formatting changes should be in another PR, it doesn't look like you touched this file other than the GitPod formatting

File: capstone-frontend/src/components/ClubPage.jsx (hunk 5 above)
Comment (antoniolinhart) [reply capstone-frontend/src/components/ClubPage.jsx:145]: +1, I think the outer div serves this purpose as it goes around everything

File: capstone-frontend/src/components/ClubPage.jsx [hunk 16]
```diff
@@ -122,46 +134,85 @@ class ClubPage extends Component {
...
-        {assignments}
-        {isOwner &&
-          <Form onSubmit={this.handleAssignmentPost} id='assignment-post-form'>
-            <Form.Group controlId='formPostAssignment'>
-              <Form.Label> Post a new assignment! </Form.Label>
-              <Form.Control as='textarea' rows='3' placeholder='Enter assignment text...' />
-            </Form.Group>
-            <Button variant='primary' type='submit'> Submit </Button>
-          </Form>
+      <div >
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/ClubPage.jsx]: Even if this is due to autoformatting, I feel like the extra space here makes it inconsistent with the other elements below, like `<Row>`

File: capstone-frontend/src/components/ClubPage.jsx [hunk 17]
```diff
@@ -122,46 +134,85 @@ class ClubPage extends Component {
...
+
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/ClubPage.jsx]: nit: Remove newline to be consistent with the formatting at the end

File: capstone-frontend/src/components/ClubPage.jsx [hunk 18]
```diff
@@ -122,46 +134,85 @@ class ClubPage extends Component {
...
+        {this.state.fetchData ?
+          (<div className="text-center mt-4">
+            <Spinner variant="primary" animation="border" role="status" />
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/ClubPage.jsx]: nit: Use single quotes for string literals

File: capstone-frontend/src/components/ClubPage.jsx [hunk 19]
```diff
@@ -122,46 +134,85 @@ class ClubPage extends Component {
...
+                  <Col className="m-auto p-0 mr-3">
+                    <div id="modal-buttons" className="mx-2">
+                      <SearchUserModal
+                        userType='viewer'
+                        text='View Members'
+                        checkoutText='Current Members'
+                        btnStyle="btn btn-primary mx-3 my-aut" />
+                    </div>
+                  </Col>}
+              </Row>
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/ClubPage.jsx]: nit: Mixture of single and double quotes throughout this file, it would be best to make them all single quotes for consistency

File: capstone-frontend/src/components/CreateList.jsx [hunk 20]
```diff
@@ -185,6 +185,7 @@ class CreateList extends Component {
                       size="lg"
                       role="status"
                       aria-hidden="true"
+                      variant='primary'
                       className="my-5"
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/CreateList.jsx]: nit: Since you added this as single quotes, you should change the rest of the props for this spinner to be single quotes - or just make it double quotes since the rest of the file is double quotes and we can change it later

File: capstone-frontend/src/components/Home.jsx [hunk 21]
```diff
@@ -0,0 +1,304 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom'
+import { CardDeck, Tabs, Tab, Card, Spinner, Button } from 'react-bootstrap';
+import ClubGridItem from './ClubGridItem.jsx'
+import { BookDescriptionOverlay } from "./BookDescriptionOverlay";
+import CreateList from './CreateList'
+import '../App.css';
+import '../styles/Home.css'
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/Home.jsx]: nit: be more consistent here with semicolons

File: capstone-frontend/src/components/Home.jsx [hunk 22]
```diff
@@ -0,0 +1,304 @@
...
+    this.fetchBookLists('shared');
+  }
+
+  fetchClubs = async (type) => {
+
+    const userID = window.localStorage.getItem('userID');
+
+    if (type === 'own') {
+      this.setState({ fetchingClubsOwned: true });
+      const clubsOwned = await fetch(`/api/clubs?ownerID=${userID}`).then(resp => resp.json());
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/Home.jsx]: Should this have a `.catch` appended to it in the case of errors?

File: capstone-frontend/src/components/Home.jsx [hunk 23]
```diff
@@ -0,0 +1,304 @@
...
+
+      let gBooks = [];
+      let bookIDs;
+
+      // Reduce to one element if on mobile
+      if (this.state.screenWidth <= 425) {
+        bookIDs = bookList.gbookIDs.slice(0, 1);
+      }
+      // Reduce to three elements if on table
+      else if (this.state.screenWidth >= 426 && this.state.screenWidth <= 768) {
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/Home.jsx:140]: nit: Since this is an **else** if, couldn't you just use the second case (`this.state.screenWidth <= 768`) because we assume that the first statement is true already?

File: capstone-frontend/src/components/Home.jsx (hunk 10 above)
Comment (antoniolinhart) [reply capstone-frontend/src/components/Home.jsx:146]: I agree! This is a really interesting way to do it I wouldn't have known about!

File: capstone-frontend/src/components/Home.jsx [hunk 24]
```diff
@@ -0,0 +1,304 @@
...
+                          <Card.Footer>
+                            <Card.Text id='booklist-card-font-size-footer' className='text-muted'>{bookList.collaboratorsIDs.length} Collaborators</Card.Text>
+                          </Card.Footer>
+                        </Card>
+                      )
+                    })}
+                  </>
+                  :
+                  <div className='text-center'>
+                    <h3 className="mt-5">No Booklists Found</h3>
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/Home.jsx]: nit: Single quotes on class names

File: capstone-frontend/src/components/Home.jsx [hunk 25]
```diff
@@ -0,0 +1,304 @@
...
+              </>
+            }
+          </Tab>
+        </Tabs>
+      </div>
+    );
+  }
+}
+
+export default Home;
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/Home.jsx]: nit: Make this a named export by putting export next to the class definition at the top

File: capstone-frontend/src/components/Home.jsx [hunk 26]
```diff
@@ -0,0 +1,304 @@
...
+
+                          <Card.Footer>
+                            <Card.Text id='booklist-card-font-size-footer' className='text-muted'>{bookList.collaboratorsIDs.length} Collaborators</Card.Text>
+                          </Card.Footer>
+                        </Card>
+                      )
+                    })}
+                  </>
+                  :
+                  <h3 className="text-center mt-5">No Shared BookLists Found</h3>}
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/Home.jsx]: nit: Use single quotes for string literals

File: capstone-frontend/src/components/Home.jsx [hunk 27]
```diff
@@ -0,0 +1,304 @@
...
+                                <Card.Body key={book.id} >
+                                  <BookDescriptionOverlay book={book}>
+                                    <Card.Img className='img-fluid my-1 light-gray-border book-img-lg' variant='top' src={book.thumbnailLink} />
+                                  </BookDescriptionOverlay>
+                                  <Card.Title id='booklist-card-book-title-font-size' className='mt-4 mb-1 text-center book-img-text-width'>{book.title}</Card.Title>
+                                </Card.Body>)}
+                            </div>
+                            :
+                            <Card.Body>
+                              <Card.Text>Booklist contains no books</Card.Text>
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/Home.jsx:313]: nit: Keep the way you refer to BookLists consistent, on line 294 you refer to them as `BookLists` but here you refer to it as a `Booklist`

File: capstone-frontend/src/components/ListPage.jsx [hunk 28]
```diff
@@ -132,6 +128,18 @@ class ListPage extends Component {
...
+                :
+                <Col className="margin-auto p-0 mr-3">
+                  <div id="modal-buttons" className="mx-2">
+                    <SearchUserModal
+                      type='booklists'
+                      userType='viewer'
+                      bookList={this.state.bookList}
+                      text='View Collaborators'
+                      checkoutText='Current Collaborators'
+                      btnStyle="btn btn-primary mx-3 h-100" />
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/ListPage.jsx]: nit: Inconsistent single quotes/double quotes

File: capstone-frontend/src/components/ListPage.jsx [hunk 29]
```diff
@@ -165,6 +173,18 @@ class ListPage extends Component {
...
+                  </div>
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/ListPage.jsx]: Same as above

File: capstone-frontend/src/components/SearchUserModal.jsx [hunk 30]
```diff
@@ -122,7 +122,7 @@ export class SearchUserModal extends Component {
       add_collaboratorsIDs: user.id,
     }
 
-    // Add Collaborator to Booklist in Firebase
+    // Remove BookList in Firebase
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/SearchUserModal.jsx]: I don't think this comment accurately describes the code it is next to

File: capstone-frontend/src/components/SearchUserModal.jsx [hunk 31]
```diff
@@ -135,7 +135,7 @@ export class SearchUserModal extends Component {
       remove_collaboratorsIDs: user.id,
     }
 
-    // Remove Collaborator to Booklist in Firebase
+    // Remove BookList in Firebase
```
Comment (antoniolinhart) [inline capstone-frontend/src/components/SearchUserModal.jsx]: Why was this changed? I feel like the original comment was more relevant

**Review by antoniolinhart — APPROVED** [review]
Looks good to me! This is a nice landing page for the user when they log in

File: capstone-frontend/src/components/ClubGridItem.jsx (hunk 2 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/ClubGridItem.jsx]: Ok.

File: capstone-backend/src/test/java/com/google/sps/servlets/SearchServletTest.java (hunk 1 above)
Comment (MauroGuerrero20) [reply capstone-backend/src/test/java/com/google/sps/servlets/SearchServletTest.java]: Ignore these changes forgot to sync with master branch.

File: capstone-frontend/src/components/ClubGridItem.jsx (hunk 2 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/ClubGridItem.jsx]: I changed it to a console.log

File: capstone-frontend/src/components/ClubGridItem.jsx (hunk 3 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/ClubGridItem.jsx:35]: Changed.

File: capstone-frontend/src/components/ClubPage.jsx (hunk 6 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/ClubPage.jsx]: Ok

File: capstone-frontend/src/components/CreateList.jsx (hunk 7 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/CreateList.jsx:227]: ```img-responsive ``` is from Bootstrap 3. ```img-fluid``` is the new alternative to ```img-responsive ```, I think it's from Bootstrap 4.

File: capstone-frontend/src/components/Home.jsx (hunk 8 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/Home.jsx]: Yes. Changed.

File: capstone-frontend/src/components/Home.jsx (hunk 12 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/Home.jsx:192]: We should discuss before removing ```MyClubs``` page.

File: capstone-frontend/src/components/ClubPage.jsx (hunk 16 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/ClubPage.jsx]: Ok

File: capstone-frontend/src/components/ClubPage.jsx (hunk 17 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/ClubPage.jsx]: Ok

File: capstone-frontend/src/components/ClubPage.jsx (hunk 18 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/ClubPage.jsx]: Ok

File: capstone-frontend/src/components/ClubPage.jsx (hunk 19 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/ClubPage.jsx]: Ok.

File: capstone-frontend/src/components/CreateList.jsx (hunk 20 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/CreateList.jsx]: Ok

File: capstone-frontend/src/components/ListPage.jsx (hunk 28 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/ListPage.jsx]: Ok

File: capstone-frontend/src/components/Home.jsx (hunk 21 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/Home.jsx]: Ok

File: capstone-frontend/src/components/Home.jsx (hunk 24 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/Home.jsx]: Ok.

File: capstone-frontend/src/components/Home.jsx (hunk 25 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/Home.jsx]: Ok

File: capstone-frontend/src/components/Home.jsx (hunk 27 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/Home.jsx:313]: Ok, I will use ```Booklists```

File: capstone-frontend/src/components/ListPage.jsx (hunk 29 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/ListPage.jsx]: Ok.

File: capstone-frontend/src/components/SearchUserModal.jsx (hunk 30 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/SearchUserModal.jsx]: This was probably a merge error I overlooked.

File: capstone-frontend/src/components/SearchUserModal.jsx (hunk 31 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/SearchUserModal.jsx]: This was probably a merge error I overlooked.

File: capstone-frontend/src/components/Home.jsx (hunk 26 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/Home.jsx]: Ok

File: capstone-frontend/src/components/Home.jsx (hunk 22 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/Home.jsx]: Ok

File: capstone-frontend/src/components/Home.jsx (hunk 23 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/Home.jsx:140]: Oh, I see. Should I change it? I feel like the current implementation is more readable. Let me know.

File: capstone-frontend/src/components/Home.jsx (hunk 12 above)
Comment (steven-solar) [reply capstone-frontend/src/components/Home.jsx:192]: Sounds good!

File: capstone-frontend/src/components/Home.jsx [hunk 32]
```diff
@@ -0,0 +1,309 @@
...
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.resize);
+  }
+
+  resize = () => {
+    this.setState({ screenWidth: window.outerWidth });
+
+    this.fetchBookLists('own');
```
Comment (VivekShankar) [inline capstone-frontend/src/components/Home.jsx:44]: Why do we need to specifically re-fetch the user's owned and shared book lists (and not their clubs) upon `resize`?

File: capstone-frontend/src/components/Home.jsx [hunk 33]
```diff
@@ -0,0 +1,309 @@
...
+    }
+    else if (type === 'shared') {
+      this.setState({ fetchingClubsMember: true });
+
+      const clubsMember = await fetch(`/api/clubs?memberIDs=${userID}`)
+        .then(resp => resp.json())
+        .catch(err => console.log(err));
+      this.setState({ clubsMember, fetchingClubsMember: false });
+    }
+  }
```
Comment (VivekShankar) [inline capstone-frontend/src/components/Home.jsx]: It looks like most of the logic in the `if` and `else` branches are the same (i.e., except the specific fetch URL). 

I'd recommend re-factoring this block to just use a conditional to set the fetch URL. The rest of the logic can be performed outside the conditional.

File: capstone-frontend/src/components/Home.jsx [hunk 34]
```diff
@@ -0,0 +1,309 @@
...
+      this.setState({ fetchingBookListsShared: true });
+
+      bookLists = await fetch(`/api/booklist?collaboratorsIDs=${userID}`)
+        .then(resp => resp.json())
+        .catch(err => console.log(err));
+
+      if (typeof bookLists === 'undefined') {
+        bookLists = [];
+      }
+    }
```
Comment (VivekShankar) [inline capstone-frontend/src/components/Home.jsx:121]: Same thing here, see above comment.

File: capstone-frontend/src/components/SearchUserModal.jsx [hunk 35]
```diff
@@ -326,25 +331,31 @@ export class SearchUserModal extends Component {
...
+                                      {this.props.removeBtnText || 'Remove'}
+                                    </Button>
+                                    :
+                                    <Button className='my-2 w-75' onClick={() => this.addUserToAddedUsers(user)}>
+                                      {this.props.addBtnText || 'Add'}
+                                    </Button>}
+                                  <br />
+                                </>}
+
+                              {user.id !== window.localStorage.getItem('userID') &&
```
Comment (VivekShankar) [inline capstone-frontend/src/components/SearchUserModal.jsx:347]: Nice check to prevent showing the friend buttons if this is the user's own card!

**Review by VivekShankar — APPROVED** [review]
LGTM - this home page packages together a lot of the cool features you've all been building over the past several weeks really nicely!

File: capstone-frontend/src/components/Home.jsx (hunk 32 above)
Comment (MauroGuerrero20) [reply capstone-frontend/src/components/Home.jsx:44]: It does a re-fetch because depending on the screen size, it will only fetch either 1, 3 or 5 books. This could probably be optimize so it doesn't do a fetch, but I think I should do that in another PR.

Comment (MauroGuerrero20) [reply capstone-frontend/src/components/Home.jsx:44]: We have to re-fetch booklists because depending on the screen size of the device it will only fetch 1, 3, or 5 books, making it compatible with multiple devices.
